_posts/frameworks/2017-04-21-transparency-with-spring-boot.md

```diff
@@ -0,0 +1,219 @@
+---
+layout: article
+title: Exposing Metrics of a Spring Boot application
+categories: [frameworks]
+modified: 2017-04-21
+author: tom
+tags: [spring, java, boot, monitoring, meter, timer, rate, moving average, transparency, metrics, actuator, dropwizard]
+comments: true
+ads: false
+image:
+  feature: 
+  teaser: teaser/measuring-tape.jpg
+  thumb:
+---
+
+How do we know if an application we just put into production is working
+as it should? How do we know that the application can cope with the
+number of users and is not slowing down to a crawl? And how do we know
+how many more users the application can handle before having to scale
+up and put another instance into the cluster? The answer to these questions
+is transparency. A good application is transparent in that it exposes
+several metrics about its health and current status that can be 
+interpreted manually as well as automatically.
+
+This post explains how to create metrics in a Java application with the Dropwizard 
+metrics library and how to expose them with Spring Boot.
+
+## What Metrics to Measure?
+
+Usual monitoring setups measure metrics like CPU, RAM and hard drive usage.
+These metrics measure the resources available to our application. These metrics
+can usually be read from the application server or operating system so that
+we don't have to do anything specific within our application to make them
+available. 
+
+These resource metrics are very important. If the monitoring setup raises an
+alarm because some resource is almost depleted, we can take action to mitigate
+that problem (i.e. adding another hard drive or putting another server into
+the load balancing cluster).
+ 
+However, there are metrics which are just as important that can only be
+created within our application: the number of payment transactions or 
+the average duration of a search in a shop application, for example. 
+These metrics give insight to the actual business value of our application
+and make capacity planning possible when held against the resource metrics.
+
+## Creating Metrics with Dropwizard
+
+Luckily, there are tools for creating such metrics, so we don't have to
+do it on our own. [Dropwizard metrics](http://metrics.dropwizard.io/) is
+such a tool, which makes it very easy to create metrics within our Java 
+application.
+
+### Injecting the MetricsRegistry
+First off, you will need a `MetricRegistry` object at which to register
+the metrics you want to measure. In a Spring Boot application, you simply
+have to add a dependency to the [Dropwizard metrics library](https://mvnrepository.com/artifact/io.dropwizard.metrics/metrics-core). 
```

Comment:
You should use a more serious page like: http://search.maven.org/#search%7Cga%7C1%7Cg%3Aio.dropwizard.metrics%20a%3Ametrics-core

or 

http://metrics.dropwizard.io/3.2.2/getting-started.html